[Payment card / Subscription] Integrate “Subscription settings” section with backend data

JKobrynski opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

To access this newly created component, paste the following link into the browser

[https://dev.new.expensify.com:8082/settings/subscription](https://dev.new.expensify.com:8082/settings/subscription)

or add this effect to `InitialSettingsPage.tsx` 

```tsx
useEffect(() => {
    Navigation.navigate(ROUTES.SETTINGS_SUBSCRIPTION)
}, [])
```

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/38627
PROPOSAL: N/A


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

1. Log into the app using an account that has a Collect/Control workspace
2. Navigate to the Subscription Settings page (see Details section above for instructions)
3. Verify if you can change subscription settings (auto renew, auto increase)
4. Verify if a feedback survey is shown if you try to disable auto renew

- [x] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

Same as **Tests** section above (it's pattern B offline behaviour)

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
5. Upload an image via copy paste
6. Verify a modal appears displaying a preview of that image
--->

Same as **Tests** section above

- [x] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Subscription settings are unavailable in the native apps

</details>

<details>
<summary>Android: mWeb Chrome</summary>

[chrome-compressed.webm](https://github.com/Expensify/App/assets/44364829/8642b2f2-61ce-4c68-a339-3d2ebad5ceb0)

</details>

<details>
<summary>iOS: Native</summary>

Subscription settings are unavailable in the native apps

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/44364829/da0ffd1e-59e2-406e-a69c-76745cf485bc

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/44364829/f0f8cce3-8f4a-4fd7-a594-cca218661c56

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/44364829/36e73dd2-dcd4-4dad-be85-deee4feba818

</details>

JKobrynski committed:
display values from API in subscription settings section

JKobrynski committed:
Merge branch 'main' into integrateSubscriptionSettingsWithBackend

JKobrynski committed:
Merge branch 'feat/subsription-settings-ui' into integrateSubscriptionSettingsWithBackend

JKobrynski committed:
add new api commands for subscription settings

JKobrynski committed:
implement auto-increase annual seats setting

JKobrynski committed:
Merge branch 'main' into implementAllToggleActions

JKobrynski committed:
start integrating UpdateSubscriptionAutoRenew API method

JKobrynski committed:
Merge branch 'main' into implementAllToggleActions

JKobrynski committed:
integrate auto renew setting and feedback survey with API

JKobrynski committed:
add offline support to subscription settings

VickyStash commented on src/types/onyx/PrivateSubscription.ts:
```diff
@@ -26,6 +27,12 @@ type PrivateSubscription = {
     /** Subscription size */
     userCount?: number;
+
+    /** Pending action */
+    pendingAction?: OnyxCommon.PendingAction;
```
Let's wrap this onyx type in `OnyxCommon.OnyxValueWithOfflineFeedback` generic instead. You can see how it's done in other onyx types.
```suggestion
```

JKobrynski replied:
Looks much better now, thanks!

VickyStash commented on src/libs/actions/Subscription.ts:
```diff
@@ -8,7 +14,100 @@ function openSubscriptionPage() {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.NVP_PRIVATE_SUBSCRIPTION,
+            value: {
+                addNewUsersAutomatically,
```
So, in a failure scenario `addNewUsersAutomatically` will be updated locally anyway?

JKobrynski replied:
Shouldn't be, good catch!

JKobrynski committed:
apply suggested changes

JKobrynski commented:
@VickyStash PR updated!

rezkiy37 commented on src/libs/actions/Subscription.ts:
```diff
@@ -8,7 +14,100 @@ function openSubscriptionPage() {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.NVP_PRIVATE_SUBSCRIPTION,
+            value: {
+                autoRenew,
```
The optimistic data already set it.
```suggestion
```

rezkiy37 commented on src/libs/actions/Subscription.ts:
```diff
@@ -8,7 +14,100 @@ function openSubscriptionPage() {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.NVP_PRIVATE_SUBSCRIPTION,
+            value: {
+                addNewUsersAutomatically,
```
The optimistic data already set it.
```suggestion
```

JKobrynski committed:
avoid overwriting values in successData

JKobrynski commented:
@VickyStash  @rezkiy37  PR updated!

DylanDylann commented on src/libs/actions/Subscription.ts:
```diff
@@ -8,7 +14,98 @@ function openSubscriptionPage() {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.NVP_PRIVATE_SUBSCRIPTION,
+            value: {
+                autoRenew,
```
Should we include disableAutoRenewReason and disableAutoRenewAdditionalNote in the optimistic data?

One more thing, It seems disableAutoRenewAdditionalNote param isn't used in anywhere, Do we intend to use this param in the future

amyevans replied:
> Not sure if we should include them as none of them is used on the frontend (they are also not in onyx as of now). Wdyt @amyevans?

I think it's fine to leave it out of the optimistic data, since as you mentioned it's not used by the frontend for anything.

>> One more thing, It seems disableAutoRenewAdditionalNote param isn't used in anywhere, Do we intend to use this param in the future
>
> Right, I removed it 😄

Hmm this is leading me to believe we have a gap in the design 😬 I will bring it up in Slack.

amyevans replied:
Alright from the discussion, we need to add an input that appears upon option selection:

![image](https://github.com/Expensify/App/assets/7277067/4b0a57e8-52b9-4722-b8d7-137f407b3b1a)

The text entered in the input gets sent to the API as the `disableAutoRenewAdditionalNote` param.

@JKobrynski do you mind handling that in this PR, or should I create a separate GH?

amyevans replied:
Per the thread we also want to make some minor copy updates:
- Update `What’s the main reason you’re disabling auto-renew on your subscription?` to `What's the main reason you're disabling auto-renew?` (es: `¿Cuál es la razón principal por la que deseas desactivar la auto-renovación?`)
- Copy for the freeform input prompt should be `What software are you moving to and why?` (es: I'm waiting on a translation and will update once I have it)

JKobrynski replied:
I'm going to do it in this PR

JKobrynski replied:
@amyevans changes applied, the only thing left is that one Spanish translation

amyevans replied:
Translation for the input prompt: `¿A qué software está migrando y por qué?`

DylanDylann commented:
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>Android: mWeb Chrome</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: Native</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>iOS: mWeb Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

<!-- add screenshots or videos here -->

</details>

<details>
<summary>MacOS: Desktop</summary>

<!-- add screenshots or videos here -->

</details>

DylanDylann commented:
@JKobrynski Auto-increase annual seats is toggled of automatically

https://github.com/Expensify/App/assets/141406735/012d0bf9-c3f1-432d-9eb4-64ab7bf630b4

JKobrynski committed:
remove disableAutoRenewAdditionalNote param from UpdateSubscriptionAutoRenewParams

JKobrynski committed:
Merge branch 'main' into implementAllToggleActions

JKobrynski commented:
@amyevans regarding [this bug](https://github.com/Expensify/App/pull/43367#issuecomment-2160964531) reported by @DylanDylann maybe you know why that's happening? These switches call different API methods, but update the same Onyx key very quickly one after another, and both values are updated correctly on the backend (it shows when you refresh the page) but for some reason the second switch snaps back to `false` state if you try to enable both quickly. Is there a way to handle this so Onyx queues that properly?

JKobrynski committed:
bring disableAutoRenewAdditionalNote back, add an input for it

amyevans commented:
Discussing [here](https://expensify.slack.com/archives/C036QM0SLJK/p1718283618694479), initial thoughts point to a possible bug in how sequential writes are being applied

amyevans commented:
Alright I created https://github.com/Expensify/App/issues/43788 for that bug, let's not block this PR on its resolution. @JKobrynski can you resolve conflicts, then @DylanDylann can you complete the checklist?

I'm assigning @blimpich to take over as internal reviewer, I am headed out on extended leave after today.

JKobrynski committed:
Merge branch 'main' into implementAllToggleActions

JKobrynski commented:
@amyevans conflicts resolved!

DylanDylann commented:
I'm working on it

DylanDylann commented on src/components/FeedbackSurvey.tsx:
```diff
@@ -57,7 +60,7 @@ function FeedbackSurvey({title, description, onSubmit, optionRowStyles}: Feedbac
         }
 
-        onSubmit(reason);
+        onSubmit(reason.key, additionalNote);
```
Let's trim `additionalNote` before submitting

DylanDylann commented:
@JKobrynski Some minor comments, everything else looks fine

DylanDylann commented:
@JKobrynski @amyevans  We are using pending action fields for APIs, then when executing APIs offline will make both Subscription settings and Subscription detail greyed out 

https://github.com/Expensify/App/assets/141406735/aa00e822-a126-48a2-ae3d-c41cdbf76f44

If we want to fix this, we can use pendingFields instead of pendingAction to specify the offline action for each part

DylanDylann commented:
About two of these above problems
**Edited: This isn't a big problem, If this is an urgent PR we can ignore this here and create an issue to follow up**

JKobrynski committed:
Merge branch 'main' into implementAllToggleActions

JKobrynski committed:
apply suggested changes

JKobrynski committed:
replace pendingAction with pendingFields

JKobrynski committed:
remove comments, add enabledWhenOffline to FeedbackSurvey submit button

JKobrynski commented:
@DylanDylann I replaced `pendingAction` with `pendingFields`, tested and it worked on my end. I also added `enabledWhenOffline` to the submit button and now it's possible to disable both switches offline. Conflicts were also resolved, let me know what you think!

DylanDylann commented on src/components/FeedbackSurvey.tsx:
```diff
@@ -57,7 +60,7 @@ function FeedbackSurvey({title, description, onSubmit, optionRowStyles}: Feedbac
         }
 
-        onSubmit(reason);
+        onSubmit(reason.key, note);
```
@JKobrynski Whoops, I mean `note.trim()` 😄

DylanDylann commented on src/pages/settings/Subscription/SubscriptionSettings/index.tsx:
```diff
@@ -66,24 +62,26 @@ function SubscriptionSettings() {
-                />
-                <Text style={[styles.mutedTextLabel, styles.mt2]}>{translate('subscription.subscriptionSettings.automaticallyIncrease')}</Text>
-            </View>
+            <OfflineWithFeedback pendingAction={privateSubscription?.pendingFields?.autoRenew ?? privateSubscription?.pendingFields?.addNewUsersAutomatically}>
```
```suggestion
            <OfflineWithFeedback pendingAction={privateSubscription?.pendingFields?.autoRenew}>
```

DylanDylann commented on src/pages/settings/Subscription/SubscriptionSettings/index.tsx:
```diff
@@ -66,24 +62,26 @@ function SubscriptionSettings() {
+                        onToggle={handleAutoRenewToggle}
+                        isActive={privateSubscription?.autoRenew ?? false}
+                    />
+                    {!!autoRenewalDate && <Text style={[styles.mutedTextLabel, styles.mt2]}>{translate('subscription.subscriptionSettings.renewsOn', {date: autoRenewalDate})}</Text>}
```
Let's add
```
<OfflineWithFeedback pendingAction={privateSubscription?.pendingFields?.addNewUsersAutomatically}>
```

DylanDylann reviewed (approved):
@JKobrynski Minor comments. Everything else looks good

JKobrynski committed:
Merge branch 'main' into implementAllToggleActions

JKobrynski committed:
apply suggested changes

JKobrynski commented:
@DylanDylann PR updated again!

DylanDylann commented:
@blimpich All yours 🚀

blimpich reviewed (approved):
Looks great! Great work everyone! 🎉

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9588292696) to staging by https://github.com/blimpich in version: 1.4.86-0 🚀

platform | result
---|---
🤖 android 🤖|failure ❌
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9588292696) to staging by https://github.com/blimpich in version: 1.4.86-0 🚀

platform | result
---|---
🤖 android 🤖|failure ❌
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9588292696) to staging by https://github.com/blimpich in version: 1.4.86-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/9619837779) to production by https://github.com/AndrewGable in version: 9.0.0-9 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
